Python/compile.c

```diff
@@ -1839,6 +1840,23 @@ compiler_add_yield_from(struct compiler *c)
     return 1;
 }
 
+static int
+compiler_pop_except_and_reraise(struct compiler *c)
+{
+    /* Stack contents
+     * [exc_info, lasti, exc]  ROT_THREE
+     * [exc, exc_info, lasti]  ROT_THREE
+     * [lasti, exc, exc_info]  POP_EXCEPT
+     * [lasti, exc]            RERAISE      1
+     */
+
+    ADDOP(c, ROT_THREE);
+    ADDOP(c, ROT_THREE);
+    ADDOP(c, POP_EXCEPT);
+    ADDOP_I(c, RERAISE, 1);
```

Comment:
Is there any possible race between these two instructions, where we switch between two async functions?  I'm guessing not but wanted to double-check.  It certainly isn't a problem with the monolithic `POP_EXCEPT_AND_RERAISE`.

(There probably is with the "pending calls" machinery, as well as the low-level "async" API, but that's not new and likely not really a problem in practice.)

Reply:
What is “pending calls”?  I’m not aware of a possibility for a race.

Reply:
There is a public C-API for request that a provided C function be called by the interpreter at some undetermined point in the future.  The eval loop checks for these basically every iteration.  So this allows arbitrary code to run between any two instructions.

However, like I said, they aren't something to worry about.

What I wanted to ask about is actually a non-issue now, so never mind. 🙂